app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -30,6 +30,42 @@ def supports_update?
       true
     end
 
+    def supports_operating_systems?
+      client.respond_to?(:operating_systems) || client.send(:client).respond_to?(:operating_systems)
+    end
+
+    def determine_os_type(host)
+      return nil unless host
+      ret = "other_linux"
+      return ret unless host.operatingsystem
+      base_name = host.operatingsystem.name =~ /redhat|centos/i ? 'rhel': host.operatingsystem.name
+      arch_name = host.architecture.name == 'x86_64' ? 'x64' : host.architecture.name if host.architecture
+
+      best_matches = available_operating_systems.select { |os| os.name.present? }.sort_by do |os|
+        rating = 0.0
+        if os.name.include?(base_name)
+          rating += 100
+          rating += (1.0/os.name.length) # prefer the shorter names a bit in case we have not found more important some specifics
+          rating += 10 if os.name.include?("#{base_name}_#{host.operatingsystem.major}")
+          rating += 10 if arch_name && os.name.include?(arch_name)
+        end
+        rating
+      end.reverse
+
+      best_matches.first.name if best_matches.first
+    end
+
+    def available_operating_systems
+      return @operating_systems if @operating_systems
+      if client.respond_to?(:operating_systems)
+        @operating_systems = client.operating_systems
+      elsif client.send(:client).respond_to?(:operating_systems)
+        @operating_systmes = client.send(:client).operating_systems
+      else
+        raise Foreman::Exception.new("Listing operating systems is not supported by the current version")
```

Comment:
shouldn't this rather return `[]`? if it's not supported by ovirt/rbovirt it shouldn't be fatal error, it's not now either